docs/studios/unity/unity-sdk/RGAction.mdx

```diff
@@ -1,116 +1,153 @@
 
-    public void Awake()
+    void Start()
     {
-        rigidbody = GetComponent<Rigidbody>();
+        _rigidbody = GetComponent<Rigidbody>();
     }
 
-    public override string GetActionName()
+    [RGAction]
+    public void Jump()
     {
-        return "MoveToPosition";
+        _rigidbody.AddForce(Vector3.up * jumpPower, ForceMode.Impulse);
     }
+}
+```
+
+</TabItem>
+<TabItem value="action_with_update" label="RGAction Starts the Action">
 
-    public override void StartAction(Dictionary<string, object> input)
+```cs
+public class Player : MonoBehavior
+{
+    public float speed = 100f;
+    public float range = 1f;
+    private RigidBody _rigidbody;
+    private Vector3? _targetPosition;
+
+    void Start()
     {
-        var targetX = float.Parse(input["x"].ToString());
-        var targetY = float.Parse(input["y"].ToString());
-        var targetZ = float.Parse(input["z"].ToString());
-        targetPosition = new Vector3(targetX, targetY, targetZ);
+        _rigidbody = GetComponent<Rigidbody>();
     }
 
     public void Update()
     {
-        
         // If we are in range, reset the action
         if (targetPosition != null && Vector3.Distance((Vector3) targetPosition, transform.position) < range)
         {
             targetPosition = null;
         }
-        
+
         // Set the target velocity
         if (targetPosition != null)
         {
-            rigidbody.velocity = ((Vector3) targetPosition - transform.position).normalized * speed * Time.deltaTime;
+            Vector3 direction = ((Vector3)targetPosition - transform.position).normalized;
+            direction.y = 0;
+            float force = speed * Time.deltaTime;
+            _rigidbody.AddForce(direction * force);
         }
     }
+
+    [RGAction]
+    public void MoveToPosition(float x, float y, float z)
+    {
+        targetPosition = new Vector3(x, y, z);
+    }
 }
 ```
+
 </TabItem>
-</Tabs>
\ No newline at end of file
+</Tabs>
+
+## Assigning Actions to GameObjects
+
+The Regression Games SDK relies on generated files to help translate commands sent from your bot into the correct actions within your Unity project.
+Whenever you add, remove, or modify actions, you will need to instruct the SDK to generate new scripts by opening the `Regression Games` dropdown and selecting "Generate Action Classes".
+
+![Generate Action Classes](img/rg-action/generate-action-classes.png)
+
+This should result in a unique generated script for each action you've defined using the `RGAction` attribute.
+You can find these files in your Unity project under `RGScripts/RGActions`.
+Each of these scripts can then be added as components to the appropriate GameObjects within your scene.
+
+![Add Action Script to GameObject](img/rg-action/add-action-script-to-game-object.png)
+
+If your bot code attempts to call an action whose script is not assigned to its GameObject,
+then the Regression Games SDK will produce an error and terminate that bot.
```

Comment:
Unless robb changed it, that isn't what used to happen.  In the past it would just ignore that action.

Reply:
Yeah I didn't touch that error handling. The only new error handling introduced would be assigning an incorrect RGAction script to a game object, because the RGAction calls a 'GetComponent' to find the expected monobehavior.